legate/driver/launcher.py

```diff
@@ -29,13 +29,6 @@
 
 __all__ = ("Launcher",)
 
-RANK_ENV_VARS = (
```

Comment:
So it looks like `LEGATE_RANK` is not set correctly in `bind.sh` currently. Note the difference between the envvars we try in that file, vs the ones we're trying here.

Consider a case where we're running on 3 nodes, with 2 ranks (process copies) per node. In that case there are 6 copies of `legate`, with global ranks (i.e. what would be stored in `OMPI_COMM_WORLD_RANK`) 0, 1, 2, 3, 4, 5 and local ranks (i.e. what would be stored in `OMPI_COMM_WORLD_LOCAL_RANK`) 0, 1, 0, 1, 0 1.

Resource binding (that is the main responsibility of `bind.sh`) cares about the local rank (e.g. each process with node local rank 0 will get the first half of the CPU cores on each node), while the naming of output files needs to differentiate between all processes on all nodes, so it wants the global rank.

Therefore, we need to get both the local and the global rank in `bind.sh`, use the local rank for all the binding happening in that file, and store the global rank in `LEGATE_RANK`, for use by the legate driver.

Reply:
038228316d3f5e3824a556eff207108650e54934 should take care of this comment